app/controllers/reservations_controller.rb

```diff
@@ -18,10 +18,12 @@ def show
 
   # POST /reservations/
   def create
-    @reservation = Reservation.create(reservation_params)
+    @ticket_desk = TicketDesk.find(params[:ticket_desk_id])
+    @reservation = @ticket_desk.reservations.create(reservation_params[:paid])
+    @ticket = @reservation.tickets.insert_all(reservation_params[:tickets])
```

Comment:
I'm not sure about this right now but I wanted to learn how to create multiple tickets for a single reservation if endpoint and its body looks like this:
```
POST /ticket_desks/1/reservations/
{
  "reservation": {
      "paid": false,
      "tickets": [
        { "price": 15, "seat": "1" },
        { "price": 15, "seat": "2" },
        { "price": 15, "seat": "3" }
      ]
  }
}
```
but it return an error `"#<ArgumentError: Empty list of attributes passed>"`

Reply:
The problem is with method `reservation_params` and with [strong params](https://github.com/rails/strong_parameters). When you sending nested parameters, you have to define nested strong parameters :)

Check how this code will work `params.require(:reservation).permit(:paid, tickets: [:price, :seat])`

Please check [this](https://github.com/rails/strong_parameters#nested-parameters) short article

Also usage of `insert_all` may not work correctly. Why? Please check this [documentation part](https://apidock.com/rails/v6.0.0/ActiveRecord/Persistence/ClassMethods/insert_all). The problem is fact that method `insert_all` will ignore all callbacks what Rails defined, so for instance, Rails will not generate automatically parameter `created_at`. You can try if I have right of course, but if there will be some problems, consider usage 

```
reservation_params[:tickets].each do |ticket_params|
  @reservation.tickets.create(ticket_params
end
```

Reply:
Ok, now I get it! Everything works fine since I implemented your adjustments:)